packages/rest/src/validation/request-body.validator.ts

```diff
@@ -142,13 +142,22 @@ function validateValueAgainstSchema(
     cache.set(schema, validatorMap);
   }
 
-  if (validate(body)) {
+  let validationResult = validate(body);
+  let validationErrors: AJV.ErrorObject[] = [];
+  try {
+    validationResult = await validationResult;
```

Comment:
It's not very friendly to change the type of `validationResult` in the same function.

Reply:
Thanks for pointing it out. I've simplified the logic. Is new one better & clear?